test/test.js

```diff
@@ -5,111 +5,598 @@ suite('GaiaHeader', function() {
   'use strict';
 
   var GaiaHeader = window['gaia-header'];
-  var realGaiaHeaderFontFit;
+  var FontFit = window['font-fit'];
 
   setup(function() {
-    this.sandbox = sinon.sandbox.create();
-    this.container = document.createElement('div');
-    document.body.appendChild(this.container);
-    this.sandbox.spy(HTMLElement.prototype, 'addEventListener');
-
-    realGaiaHeaderFontFit = window['./lib/font-fit'];
-    window['./lib/font-fit'] = window['./test/mocks/mock_font_fit'];
+    var windowWidth = 500;
+    this.sinon = sinon.sandbox.create();
+
+    // Spy/stub FontFit
+    this.sinon.spy(GaiaHeader.prototype, 'fontFit');
+    this.fontFit = GaiaHeader.prototype.fontFit; // alias
+
+    this.sinon.spy(HTMLElement.prototype, 'addEventListener');
+    this.sinon.spy(GaiaHeader.prototype, 'setTitleStyle');
+    this.sinon.spy(GaiaHeader.prototype, 'runFontFit');
+    this.runFontFit = GaiaHeader.prototype.runFontFit;
+
+    // DOM container to put test cases
+    this.dom = document.createElement('div');
+    this.dom.style.width = windowWidth + 'px';
+    document.body.appendChild(this.dom);
+
+    // Overwrite `window.innerWidth` so we know
+    // it's the same no matter the size of the
+    // browser window across test environments.
+    this.innerWidth = Object.getOwnPropertyDescriptor(window, 'innerWidth');
+    this.innerWidth_spy = sinon.spy(function() { return windowWidth; });
```

Comment:
you can define the function like this: `() => windowWidth` which is shorter, but up to you